src/axom/primal/operators/winding_number.hpp

```diff
@@ -444,6 +458,275 @@ int winding_number(const Point<T, 3>& query,
+
+      auto matx = numerics::Matrix<T>::zeros(3, 3);
+
+      matx(0, 0) = x * x * C + c;
+      matx(0, 1) = x * y * C - z * s;
+      matx(0, 2) = x * z * C + y * s;
+
+      matx(1, 0) = y * x * C + z * s;
+      matx(1, 1) = y * y * C + c;
+      matx(1, 2) = y * z * C - x * s;
+
+      matx(2, 0) = z * x * C - y * s;
+      matx(2, 1) = z * y * C + x * s;
+      matx(2, 2) = z * z * C + c;
+
+      return matx;
+    };
+
+    // Lambda to rotate the input point using the provided rotation matrix
+    auto rotate_point = [&query](const numerics::Matrix<T>& matx,
+                                 const Point<T, 3> input) -> Point<T, 3> {
+      Vector<T, 3> shifted(query, input);
+      Vector<T, 3> rotated;
+      numerics::matrix_vector_multiply(matx, shifted.data(), rotated.data());
+      return Point<T, 3>(
+        {rotated[0] + query[0], rotated[1] + query[1], rotated[2] + query[2]});
+    };
+
+    // Find vector from query to the bounding box
+    Point<T, 3> closest = closest_point(query, oBox);
+    Vector<T, 3> v0 = Vector<T, 3>(query, closest).unitVector();
+
+    // Find the direction of a ray perpendicular to that
+    Vector<T, 3> v1;
+    if(axom::utilities::isNearlyEqual(v0[0], v0[1], EPS))
+      v1 = Vector<T, 3>({v0[2], v0[2], -v0[0] - v0[1]}).unitVector();
+    else
+      v1 = Vector<T, 3>({-v0[1] - v0[2], v0[0], v0[0]}).unitVector();
+
+    // Rotate v0 around v1 until it is perpendicular to the plane spanned by k and v1
+    double ang = (v0[2] < 0 ? 1.0 : -1.0) *
+      acos(-(v0[0] * v1[1] - v0[1] * v1[0]) / sqrt(v1[0] * v1[0] + v1[1] * v1[1]));
+    auto rotator = angleAxisRotMatrix(ang, v1);
+
+    // Collect rotated curves into the curved Polygon
+    // Set up the (0, v) and (1, v) isocurves, rotated
+    boundingPoly[0].setOrder(ord_v);
+    boundingPoly[2].setOrder(ord_v);
+    if(patchIsRational)
+    {
+      boundingPoly[0].makeRational();
+      boundingPoly[2].makeRational();
+    }
+    for(int q = 0; q <= ord_v; ++q)
+    {
+      boundingPoly[0][q] = rotate_point(rotator, bPatch(ord_v, q));
+      boundingPoly[2][q] = rotate_point(rotator, bPatch(0, ord_v - q));
+
+      if(patchIsRational)
+      {
+        boundingPoly[0].setWeight(q, bPatch.getWeight(ord_v, q));
+        boundingPoly[2].setWeight(q, bPatch.getWeight(0, ord_v - q));
+      }
+    }
+
+    // Set up the (u, 0) and (u, 1) isocurves
+    boundingPoly[1].setOrder(ord_u);
+    boundingPoly[3].setOrder(ord_u);
+    if(patchIsRational)
+    {
+      boundingPoly[1].makeRational();
+      boundingPoly[3].makeRational();
+    }
+    for(int p = 0; p <= ord_u; ++p)
+    {
+      boundingPoly[1][p] = rotate_point(rotator, bPatch(ord_u - p, ord_u));
+      boundingPoly[3][p] = rotate_point(rotator, bPatch(p, 0));
+
+      if(patchIsRational)
+      {
+        boundingPoly[1].setWeight(p, bPatch.getWeight(ord_u - p, ord_u));
+        boundingPoly[3].setWeight(p, bPatch.getWeight(p, 0));
+      }
+    }
```

Comment:
I agree that BezierPatch should be able to return a `CurvedPolygon` (or `Array<BezierCurve>`), especially as we more forward to the trimmed NURBS patch case, where there are more than just the 4 isocurves. That way, the interface could be pretty uniform.

As far as adding in those rotations go, it would be necessarily less efficient, since you'd have to store the intermediates. However, if that's just a cost we're accepting, then I think it would also be nice for a `BezierCurve` or `BezierPatch` object to just have a `.rotation( rotation_matrix/quaternion )` method. A `.rotation( axis, angle )` method would also work, but would likely be less efficient. You'd need to either apply it to the whole patch and rotate unnecessary internal nodes, or apply it to each curve and construct the rotation matrix/quaternion 4 times. 

Either way, I think the current implementation is a little ugly looking, but I think it's clear enough for now. Maybe if we add a quaternion class down the line we can consider it?

Reply:
I agree -- let's leave it as is for now and perhaps revisit in the future